ClaudiaRojasSoto/Blog_App
# Feature/authorizations

## Description
This pull request aims to implement authorization using CanCanCan as per the provided requirements. The project includes the addition of a 'role' column to the users table, and the authorization logic for post and comment deletion based on user roles.

## General Requirements
- Linter errors have been resolved.
- Gitflow has been followed.
- Documentation is provided in a professional manner.

## Ruby Requirements
- Best practices for Ruby have been adhered to.

## Project Requirements
### 1. Install CanCanCan
CanCanCan has been successfully installed in the project.

### 2. Add 'role' Column
A 'role' column has been added to the users table using a migration.

### 3. Post Deletion Authorization
Users can delete a post if it belongs to them or if they have an 'admin' role. CanCanCan is used for this authorization. The "Delete" button is displayed only to authorized users.

### 4. Comment Deletion Authorization
Users can delete a comment if it belongs to them or if they have an 'admin' role. CanCanCan is used for this authorization. The "Delete" button is displayed only to authorized users.

## Changes Made
- Added CanCanCan gem to Gemfile.
- Created a migration to add the 'role' column to the users table.
- Implemented authorization logic for post and comment deletion.
- Modified the view to show the "Delete" button to authorized users only.

## Files Modified
1. `Gemfile`
2. `db/migrate/<migration_filename>`
3. `app/controllers/posts_controller.rb`
4. `app/controllers/comments_controller.rb`
5. `app/views/posts/show.html.erb`
6. `app/views/comments/show.html.erb`

## How to Test
1. Make sure you have CanCanCan installed.
2. Run the migration to add the 'role' column to the users table.
3. Test the post and comment deletion functionality as a regular user and as an 'admin' user.
4. Verify that the "Delete" button is displayed only to authorized users.

## Additional Notes
Please review this pull request thoroughly to ensure that all requirements have been met. Feel free to ask for any clarifications or further assistance if needed and provide us feedback to improve the app. 

Thank you for your attention to detail and professionalism.

Sincerely,
@Dachrono 
@ClaudiaRojasSoto
